packages/react-native-session-replay/android/src/main/kotlin/com/datadog/reactnative/sessionreplay/extensions/ReactDrawablesExt.kt

```diff
@@ -0,0 +1,194 @@
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.reactnative.sessionreplay.extensions
+
+import android.content.res.Resources
+import android.graphics.Bitmap
+import android.graphics.Bitmap.Config
+import android.graphics.Canvas
+import android.graphics.drawable.BitmapDrawable
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.ShapeDrawable
+import android.graphics.drawable.VectorDrawable
+import android.widget.ImageView
+import androidx.appcompat.graphics.drawable.DrawerArrowDrawable
+import com.facebook.drawee.drawable.ArrayDrawable
+import com.facebook.drawee.drawable.ForwardingDrawable
+import com.facebook.drawee.drawable.RoundedBitmapDrawable
+import com.facebook.drawee.drawable.ScaleTypeDrawable
+import com.facebook.drawee.drawable.ScalingUtils
+
+internal fun ScaleTypeDrawable.imageViewScaleType(): ImageView.ScaleType? {
+    return when (this.scaleType) {
+        ScalingUtils.ScaleType.CENTER -> ImageView.ScaleType.CENTER
+        ScalingUtils.ScaleType.CENTER_CROP -> ImageView.ScaleType.CENTER_CROP
+        ScalingUtils.ScaleType.CENTER_INSIDE -> ImageView.ScaleType.CENTER_INSIDE
+        ScalingUtils.ScaleType.FIT_CENTER -> ImageView.ScaleType.FIT_CENTER
+        ScalingUtils.ScaleType.FIT_START -> ImageView.ScaleType.FIT_START
+        ScalingUtils.ScaleType.FIT_END -> ImageView.ScaleType.FIT_END
+        ScalingUtils.ScaleType.FIT_XY -> ImageView.ScaleType.FIT_XY
+        else -> null
+    }
+}
+
+internal fun ArrayDrawable.getScaleTypeDrawable(): ScaleTypeDrawable? {
+    for (i in 0 until this.numberOfLayers) {
+        try {
+            (this.getDrawable(i) as? ScaleTypeDrawable)?.let {
+                return it
+            }
+        } catch(_: IllegalArgumentException) { }
+    }
+
+    return null
+}
+
+internal fun ArrayDrawable.getDrawableOrNull(index: Int): Drawable? {
+    return try {
+        this.getDrawable(index)
+    } catch (_: IllegalArgumentException) {
+        null
+    }
+}
+
+internal fun ForwardingDrawable.tryToExtractBitmap(resources: Resources): Bitmap? {
+    val forwardedDrawable = this.drawable
+    return if (forwardedDrawable != null) {
+        forwardedDrawable.tryToExtractBitmap(resources)
+    } else {
+        this.toBitmapOrNull(
+            this.intrinsicWidth,
+            this.intrinsicHeight,
+            Bitmap.Config.ARGB_8888
+        )
+    }
+}
+
+internal fun RoundedBitmapDrawable.tryToExtractBitmap(): Bitmap? {
+    val privateBitmap = try {
+        val field = RoundedBitmapDrawable::class.java.getDeclaredField("mBitmap")
+        field.isAccessible = true
+        field.get(this) as? Bitmap
+    } catch (_: NoSuchFieldException) {
+        null
+    } catch (_: IllegalAccessException) {
+        null
+    } catch (_: Exception) {
+        null
+    }
+
+    return privateBitmap ?: this.toBitmapOrNull(
+        this.intrinsicWidth,
+        this.intrinsicHeight,
+        Bitmap.Config.ARGB_8888
+    )
```

Comment:
```suggestion
    return privateBitmap ?: toBitmapOrNull(
        intrinsicWidth,
        intrinsicHeight,
        Bitmap.Config.ARGB_8888
    )
```

same for other methods - you can remove `this`, it is implicit